### PR TITLE
Feat winner

### PR DESCRIPTION
**Pull request by ghost**
- 5 commits developed during classes (incl. Leading functionality)
- 2 commits adding Winner functionality

### CODE REVIEW

File: pkg/skullking/round.go [hunk 1]
```diff
@@ -37,7 +46,8 @@ func (t *Trick) Play(p Play) error {
 
 // Winner will return the player that wins the current Trick
 func (t *Trick) Winner() *Player {
-	return nil
+	winnerPosition := t.Winner2()
```
Comment (MetalBlueberry) [inline pkg/skullking/round.go]: I don't think winner2 is a good name for a function.

File: pkg/skullking/round.go [hunk 2]
```diff
@@ -46,5 +56,89 @@ func (t *Trick) Points() int {
...
+		t.Table[position].Card.Type != CardTypeSuitBlack &&
+		t.Table[position].Card.Type != CardTypeSuitYellow &&
+		t.Table[position].Card.Type != CardTypeSuitPurple {
+		return CardTypeNone
+	}
+
+	return t.Table[position].Card.Type
+}
+
+func (t *Trick) GatheringInfo() Info {
```
Comment (MetalBlueberry) [inline pkg/skullking/round.go]: This function should be internal. Make the first letter lowercase

File: pkg/skullking/deck.go [hunk 3]
```diff
@@ -35,9 +35,13 @@ const (
...
 
 	CardTypeSuitParrot        CardType = "Parrot"
+	CardTypeSuitGreen         CardType = "green"
 	CardTypeSuitTreasureChest CardType = "Treasure Chest"
-	CardTypeSuitJollyRoger    CardType = "Jolly Roger"
+	CardTypeSuitYellow        CardType = "yellow"
 	CardTypeSuitPirateMap     CardType = "Pirate Map"
+	CardTypeSuitPurple        CardType = "purple"
+	CardTypeSuitJollyRoger    CardType = "Jolly Roger"
+	CardTypeSuitBlack         CardType = "black"
```
Comment (MetalBlueberry) [inline pkg/skullking/deck.go]: I don't think colors and suit names behave the same. For example, Black and JollyRoger should be the same, but they are not.

Maybe we can consider dropping the names and leave only the colors

Comment (ghost) [reply pkg/skullking/deck.go]: I just deleted every instance of the old naming convention

File: pkg/skullking/round.go (hunk 1 above)
Comment (ghost) [reply pkg/skullking/round.go]: Winner2 was meant only as a WIP, now it's called WinnerPosition

**Review by MetalBlueberry — APPROVED** [review]

**Pull request merged**